hugo: add tags

dvdksn opened the pull request:
Adds a "Categories" taxonomy to docs, which lets us define content relationships based on domains or content types, and surface that as an alternative means of navigating between pages.

stephaurelio reviewed (commented):
I like the update of removing **Categories** from the top-level nav so there isn't a secondary IA to navigate. I think we should keep the **Section** column on the category main page because it does help orient where the content lives (in the last sync we discussed removing this).

On mobile, the table of contents plus the categories before the content starts could be a little overwhelming to scroll through (depending on the article) - curious how the team thinks about the mobile experience! I wonder if the ToC could be collapsible?

One thing I think we should have in place before adding categories is a source-of-truth for category definitions (either in Notion, or if we add to our contribution guide anything about categories) to make sure the category content is consistent.

craig-osterhout commented:
It looks good to me. Ran into one issue.
In my browser, I can't fully scroll down to categories on https://deploy-preview-19665--docsdocker.netlify.app/desktop/release-notes/

Also, I remember someone mentioned this at the offsite. The FAQ and Samples categories seem redundant with the top nav. It seems like we should remove FAQ and Samples items from the top nav and move those topics back into the manuals/references.

Lastly, the original implementation was alternate navigation. With the top nav item removed, users will be down in a topic before they can even access/browse by the categories. At that point, they've already browsed and are probably on the topic they want. Maybe add something on the homepage so we have, "browse by section" and "browse by content type". And on that note, I'm thinking whether or not displaying these particular categories on pages are even helpful to users. I think they're only helpful if there's a use-case where users want to go from one page to other pages in that content type. For example, if I went to the Docker Desktop Windows installation page and now I want to see all other product installation pages. I can't think of a use-case for that.

dvdksn commented:
> And on that note, I'm thinking whether or not displaying these particular categories on pages are even helpful to users.

I agree very much with this sentiment for multiple of these categories:

- installation
- quickstart
- samples
- faq

These pages are either already logically grouped, or they're interesting for cross-product navigation. At the same time I can think of other terms that could be interesting:

- environment variables
- secrets
- github actions (or ci)
- gpu (?)
- deployment (?)

dvdksn commented:
I've changed it up a bit. I've removed the categories I had doubts about (as expressed above). I also changed it from "Categories" (vague/unclear term) to "tags".

@stephaurelio to your point about source of truth, I've added an error when you try to use a tag that isn't defined
<img width="875" alt="image" src="https://github.com/docker/docs/assets/35727626/a9a235cb-97c6-43df-a069-80b1ea58d785">

usha-mandya reviewed (approved):
LGTM. @dvdksn Could you resolve the conflicts?

dvdksn commented:
@usha-mandya rebased

usha-mandya reviewed (approved):
:shipit:
